### DISCUSSION TITLE
Is there a way to install irobot-create-msgs on my ROS Humble container for the NVIDIA Jetson Orin Nano?

### DISCUSSION THREAD

**Discussion by johnladolfo** [iRobotEducation/create3_docs#426] [Questions] [answered]
### How are you connecting to your Create 3?

USB Ethernet (ROS 2)

### Computer(s) Model(s) and Operating System(s)

NVIDIA Jetson Orin Nano, Ubuntu 20.04

### Which version of ROS 2 is installed on your computer?

Humble

### Which firmware version is installed on your robot?

H.2.1

### Which RMW is your robot running?

CycloneDDS

### Is the robot connected to a network? If so, what is the network type?

Yes; home network

### Are there multiple Create 3 robots connected to your network?

No

### Is multicast enabled?

None

### What is the Adapter Board's USB/BLE Toggle currently switched to?

USB

### Describe your question.

I'm currently using the NVIDIA Isaac ROS Humble container on my Orin Nano to communicate with the Create3. The problem is that I can't install the ros-humble-irobot-create-msgs package into my container since the machine runs on Ubuntu 20.04. Because of this, I can't use any of the prebuilt API commands and I get the error:

The passed ____ type is invalid

I've tried installing the galactic version of the package inside the container instead, but when I try to run the API commands after sourcing the galactic setup.bash, I get this error: 

Failed to load entry point 'pub': /opt/ros/humble/install/lib/librcl.so: undefined symbol: rmw_subscription_content_filter_options_init
The C extension '/opt/ros/humble/install/lib/python3.8/site-packages/rclpy/_rclpy_pybind11.cpython-38-aarch64-linux-gnu.so' failed to be imported while being present on the system. Please refer to 'https://index.ros.org/doc/ros2/Troubleshooting/Installation-Troubleshooting/#import-failing-even-with-library-present-on-the-system' for possible solutions
Traceback (most recent call last):
  File "/opt/ros/humble/install/bin/ros2", line 33, in <module>
    sys.exit(load_entry_point('ros2cli==0.18.6', 'console_scripts', 'ros2')())
  File "/opt/ros/humble/install/lib/python3.8/site-packages/ros2cli/cli.py", line 50, in main
    add_subparsers_on_demand(
  File "/opt/ros/humble/install/lib/python3.8/site-packages/ros2cli/command/__init__.py", line 250, in add_subparsers_on_demand
    extension.add_arguments(
  File "/opt/ros/humble/install/lib/python3.8/site-packages/ros2topic/command/topic.py", line 29, in add_arguments
    add_subparsers_on_demand(
  File "/opt/ros/humble/install/lib/python3.8/site-packages/ros2cli/command/__init__.py", line 237, in add_subparsers_on_demand
    extension = command_extensions[name]
KeyError: 'pub'

Is there a way I can easily obtain the irobot-create-msgs package in my container? The Create3 docs mentioned I could use the NVIDIA Isaac ROS containers but did not provide a way to install this package within it.

**Comment by johnladolfo** [comment, marked as answer]
I cloned the humble branch of the irobot-create-msgs repo into my container workspace, built it using colcon build, sourced the overlay, and that seemed to resolve the issue and let me use the API commands. This method seems to be a good solution, but if anyone has any other suggestions, I would greatly appreciate it.

**Reply by alsora** [reply]
If you are using an Ubuntu 20.04 container, building the humble branch of the repo is the only correct solution.
ROS 2 Humble debian packages are only available for Ubuntu 22.04
